src/render_engine/vulkan/vulkan_render_engine.cpp

```diff
@@ -1391,17 +1443,30 @@ namespace nova::renderer::rhi {
         vkGetDeviceQueue(device, copy_family_idx, 0, &copy_queue);
     }
 
-    bool VulkanRenderEngine::does_device_support_extensions(VkPhysicalDevice device) {
+    bool VulkanRenderEngine::does_device_support_extensions(VkPhysicalDevice device, const std::vector<char*>& required_device_extensions) {
         uint32_t extension_count;
         vkEnumerateDeviceExtensionProperties(device, nullptr, &extension_count, nullptr);
         std::vector<VkExtensionProperties> available(extension_count);
         vkEnumerateDeviceExtensionProperties(device, nullptr, &extension_count, available.data());
 
-        std::set<std::string> required = {VK_KHR_SWAPCHAIN_EXTENSION_NAME};
+        std::set<std::string> required;
```

Comment:
Is the use of std::set mandatory? It's a pretty butt structure if you don't need it.

Reply:
I guess it isn't mandatory but it's less code than `std::remove_if`

Reply:
I was just saying why use set over `std::unordered_set`

Reply:
Because I didn't realize `std::unordered_set` was a thing

Fixed now